haystack_experimental/core/super_component/super_component.py

```diff
@@ -2,40 +2,108 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-from typing import Any, Dict, List, Optional, Tuple
+import functools
+from types import new_class
+from typing import Any, Dict, List, Optional, Tuple, Union
 
-from haystack import Pipeline, component
+from haystack import Pipeline, component, logging
+from haystack.core.pipeline.async_pipeline import AsyncPipeline
 from haystack.core.pipeline.utils import parse_connect_string
 from haystack.core.serialization import default_from_dict, default_to_dict, generate_qualified_class_name
 
 from haystack_experimental.core.super_component.utils import _delegate_default, _is_compatible
 
+logger = logging.getLogger(__name__)
+
+
+class InvalidMappingTypeError(Exception):
+    """Raised when input or output mappings have invalid types or type conflicts."""
+
+    pass
+
+
+class InvalidMappingValueError(Exception):
+    """Raised when input or output mappings have invalid values or missing components/sockets."""
 
-class InvalidMappingError(Exception):
-    """Raised when input or output mappings are invalid or type conflicts are found."""
     pass
 
 
 @component
-class SuperComponent:
+class _SuperComponent:
     """
     A class for creating super components that wrap around a Pipeline.
 
     This component allows for remapping of input and output socket names between the wrapped pipeline and the
     SuperComponent's input and output names. This is useful for creating higher-level components that abstract
     away the details of the wrapped pipeline.
-    """
 
+    ### Usage example
```

Comment:
It's great that we don't cause any breaking change in this usage example. We just need to move it to `SuperComponent` instead of `_SuperComponent`.